tests/test_date.py

```diff
@@ -297,12 +438,97 @@ def test_should_parse_with_no_break_space_in_dates(self):
         param(['ur', 'li']),
-        param(['pk',]),
+        param(['pk', ]),
     ])
     def test_should_raise_error_when_unknown_language_given(self, shortnames):
         with self.assertRaisesRegexp(ValueError, '%r' % ', '.join(shortnames)):
             date.DateDataParser(languages=shortnames)
 
+    def when_multiple_dates_are_parsed(self):
+        self.results = []
+        for date_string in self.date_strings:
+            try:
+                result = self.parser.get_date_data(date_string)
+            except Exception as error:
+                result = error
+            finally:
+                self.results.append(result)
+
+    def given_multiple_dates(self, date_to_parse):
+        self.date_strings = date_to_parse.keys()
+        self.dates = list(date_to_parse.values())
+
+    def given_known_languages(self, *shortnames):
+        self.ordered_languages = OrderedDict()
+
+        self.known_languages = [self.language_map.get(shortname)
+                                for shortname in shortnames]
+
+        for language in self.known_languages:
+            self.ordered_languages[language.shortname] = language
+
+        self.language_loader._data = self.ordered_languages
+
+    def given_empty_languages_parser(self):
+        self._data = {}
+        self.parser = date.DateDataParser(allow_redetect_language=True)
+
+        language_loader = LanguageDataLoader()
+        self.language_map = date.default_language_loader.get_language_map()
+        self.language_loader = language_loader
+        language_loader._load_data = MethodType(self.given_empty_languages_parser, language_loader)
+        self.add_patch(patch('dateparser.date.default_language_loader', new=language_loader))
+
+    def given_date_string(self, date_string):
+        self.date_string = date_string
+
+    def given_date_format(self, date_format):
+        self.date_format = date_format
+
+    def given_expected_date(self, expected_date):
+        self.expected_date = expected_date
+
+    def given_today(self):
+        self.today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
+
+    def when_date_string_is_parsed(self):
+        self.date_data = self.parser.get_date_data(self.date_string)
+
+    def when_date_data_is_parsed(self):
+        self.date_data = self.parser.get_date_data(self.date_string)
+
+    def when_should_parse_date_with_timezones_using_format(self):
+        self.date_data = self.parser.get_date_data(self.date_string, date_formats=[self.date_format])
+
+    def then_date_was_parsed(self):
+        self.assertIsNotNone(self.date_data['date_obj'])
+
+    def then_check_yesterday(self):
+        self.assertIsNotNone(self.date_data['date_obj'],
+                             "could not parse date from: %s" % self.date_string)
+
+    def then_date_is_n_days_ago(self, days):
+        self.assertIsNotNone(self.date_data['date_obj'])
+        self.days = self.today.date() - self.date_data['date_obj'].date()
```

Comment:
I must have misunderstood your example, I thought the days_ago was a param to pass into that method?